src/libs/web3/web3.utils.ts

```diff
@@ -16,6 +19,28 @@ const { type } = parser.getDevice();
 
 export const isMobile = type === 'mobile' || type === 'tablet';
 
+/**
+ * Get the injected wallet provider in window
+ * Returns window.ethereum if window.ethereum exists, window.ethereum[flag], is true and window.ethereum is not metamask
+ * Otherwise, returns window[walletObject]
+ *
+ * @param flag name of the flag in window.ethereum to check. If undefined, will not check this flag and default to walletObject
+ * @param walletObject name of the object in window that the injected provider will populate
+ * @returns Injected provider in wallet or undefined if not present
+ */
+export function getInjectedProvider(walletObject: string, flag?: string) {
```

Comment:
This method is a bit messy in the sense that it both gets info from the caller but also makes some attempts on the ethereum object. I'd simply go with each of the current callers using their own logic to get the injected provider. If you separate the connectors into their own modules or classes, each with its implementation then it becomes more tidy and clear and can be used in a generic way. Since most do a similar things you can offer a method that's initialized with the expected object name.

This method here is also implicitly dictating who wins. If there's ethereum and no flag it's alway ethereum.

Reply:
This method follows the methodology used by Rainbow where the window.ethereum object is the default as specified in [EIP-1193](https://eips.ethereum.org/EIPS/eip-1193).

Agree with separating the EIP1193 connectors' injected provider retrieval logic though, will do that for the wagmi integration